spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/Linear.scala

```diff
@@ -138,8 +136,8 @@ class Linear(
     updateWithNewTensor(updateOutputTensors, 0, input)
 
     if (isTraining()) {
-      weight.syncToNative()
-      bias.syncToNative()
+      weight.sync()
```

Comment:
if not training, when to sync weight and bias to native?

Reply:
the previous commit will sync weight and bias at `setMemoryData`. Have pushed a new commit which will sync weight hard code when compiling.